src/runtime_src/core/common/api/aie/xrt_graph.cpp

```diff
@@ -109,7 +109,50 @@ class graph_impl
     device->read_graph_rtp(handle, port, buffer, size);
   }
 };
+}
+namespace xrt {
+namespace aie {
+
+class event_impl
+{
+private:
+  std::shared_ptr<xrt_core::device> device;
+  int handle;
+
+public:
+  event_impl(std::shared_ptr<xrt_core::device> dev, int phdl)
+    : device(std::move(dev))
+    , handle(phdl)
+  {}
+
+  ~event_impl()
+  {}
+
+  uint64_t
+  read_profiling()
+  {
+    return device->read_profiling(handle);
+  }
+
+  void
+  stop_profiling()
+  {
+    device->stop_profiling(handle);
+  }
+
+  int getHandle() const { return handle; }
+};
+
+}
+
+}
 
+int get_event_handle(const std::shared_ptr<xrt::aie::event_impl>& event_ptr) {
+    if (event_ptr) {
+        return event_ptr->getHandle();
+    } else {
+        return -1;
+    }
 }
```

Comment:
Is -1 an error, if so throw?  See, I asking because there are no comments to help me :-)
Also reformat:
```suggestion
int 
get_event_handle(const std::shared_ptr<xrt::aie::event_impl>& event_ptr) 
{
  if (auto hdl = event_ptr->get_handle())
     return hdl;
  
  throw ???;
}
```

Reply:
sure